src/Microsoft.OData.Core/ODataReaderCore.cs

```diff
@@ -372,6 +384,84 @@ public override sealed Task<bool> ReadAsync()
         }
 #endif
 
+        /// <summary>
+        /// Creates a stream for reading an inline stream property.
+        /// </summary>
+        /// <returns>A stream for reading the stream property.</returns>
+        public override sealed Stream CreateReadStream()
+        {
+            if (this.State == ODataReaderState.Stream)
+            {
+                StreamScope scope = this.CurrentScope as StreamScope;
+                Debug.Assert(scope != null, "ODataReaderState.Stream when Scope is not a StreamScope");
+                if (scope.StreamingState != StreamingState.None)
+                {
+                    throw new ODataException(Strings.ODataReaderCore_CreateReadStreamCalledInInvalidState);
+                }
+
+                scope.StreamingState = StreamingState.Streaming;
+                return new ODataNotificationStream(this.InterceptException(this.CreateReadStreamImplementation), this);
+            }
+            else
+            {

```

Comment:
Can remove "else" and throw new ODataException(...) directly

Reply:
Maybe we can do "

if (this.State != ODataReaderState.Stream)
{  throw...}

// do other things...

---
In reply to: [268359523](https://github.com/OData/odata.net/pull/1204#discussion_r268359523) [](ancestors = 268359523)

Reply:
sure; i like it.

---
In reply to: [268359707](https://github.com/OData/odata.net/pull/1204#discussion_r268359707) [](ancestors = 268359707,268359523)